File: exercism/clojure/difference-of-squares/src/difference_of_squares.clj
(ns difference-of-squares)

(defn sum-of-squares [upper-bound]
  (reduce + (map #(* %1 %1) (range 1 (+ 1 upper-bound)))))

(defn square-of-sum [upper-bound]
  (int (Math/pow (reduce + (range 1 (+ 1 upper-bound))) 2)))

(defn difference [upper-bound]
  (- (square-of-sum upper-bound) (sum-of-squares upper-bound)))

File: exercism/dart/difference-of-squares/lib/difference_of_squares.dart
import 'dart:math';

class DifferenceOfSquares {
  int _square(int number) => pow(number, 2).toInt();
  
  int _sum(int a, int b) => a + b;
  
  List<int> _generateList(int max) => List<int>.generate(max, (i) => i + 1);
  
  int squareOfSum(int max) {
    return _square(_generateList(max).reduce(_sum));
  }

  int sumOfSquares(int max) {
    return _generateList(max).map(_square).reduce(_sum);
  }

  int differenceOfSquares(int max) {
    return squareOfSum(max) - sumOfSquares(max);
  }
}
